vault/data_source_ldap_static_role_credentials.go

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/vault/api"
+)
+
+func ldapStaticCredDataSource() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: ReadContextWrapper(readLDAPStaticCreds),
+		Schema: map[string]*schema.Schema{
+			consts.FieldMount: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "LDAP Secret Backend to read credentials from.",
+			},
+			consts.FieldRoleName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the role.",
+			},
+			consts.FieldDN: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Distinguished name (DN) of the existing LDAP entry to manage password rotation for.",
+			},
+			consts.FieldLastVaultRotation: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Last time Vault rotated this static role's password.",
+			},
+			consts.FieldPassword: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Password for the static role.",
+				Sensitive:   true,
+			},
+			consts.FieldLastPassword: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Last known password for the static role.",
+				Sensitive:   true,
+			},
+			consts.FieldRotationPeriod: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "How often Vault should rotate the password of the user entry.",
+			},
+			consts.FieldTTL: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Duration in seconds after which the issued credential should expire.",
+			},
+			consts.FieldUsername: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Name of the static role.",
+			},
+		},
+	}
+}
+
+func readLDAPStaticCreds(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	mount := d.Get(consts.FieldMount).(string)
+	role := d.Get(consts.FieldRoleName).(string)
+	fullPath := fmt.Sprintf("%s/static-cred/%s", mount, role)
+
+	secret, err := client.Logical().ReadWithContext(ctx, fullPath)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error reading from Vault: %s", err))
+	}
+	log.Printf("[DEBUG] Read %q from Vault", fullPath)
+	if secret == nil {
+		return diag.FromErr(fmt.Errorf("no role found at %q", fullPath))
+	}
+
+	response, err := parseLDAPStaticCredSecret(secret)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(response.username)
+	d.Set(consts.FieldDN, response.dn)
```

Comment:
We should probably handle any errors returned from the `Set` operations.

Also nit: is there a way we can do the error handling in a loop like we did [here](https://github.com/hashicorp/terraform-provider-vault/blob/VAULT-15208/ldap-static-credential-data-source/vault/resource_ldap_secret_backend_static_role.go#L110-L116)?

Reply:
There doesn't seem to be an obvious way, since some fields are of different types and/or need to be checked for existence.

Reply:
Ah cool no problem, thanks for looking into it!